src/core/internal/dassert.d

```diff
@@ -1,18 +1,62 @@
 /*
-Provides light-weight formatting utilities for pretty-printing
-on assertion failures
-*/
+ * Support for rich error messages generation with `assert`
+ *
+ * This module provides the `_d_assert_fail` hooks which are instantiated
+ * by the compiler whenever `-checkaction=context` is used.
+ * There are two hooks, one for unary expressions, and one for binary.
+ * When used, the compiler will rewrite `assert(a >= b)` as
+ * `assert(a >= b, _d_assert_fail!"<"(a, b))`, inverting the operator.
+ * Temporaries will be created to avoid side effects if deemed necessary
+ * by the compiler.
+ *
+ * For more information, refer to the implementation in DMD frontend
+ * for `AssertExpression`'s semantic analysis.
+ *
+ * Copyright: D Language Foundation 2018 - 2020
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:    $(LINK2 https://github.com/dlang/druntime/blob/master/src/core/internal/dassert.d, _dassert.d)
+ * Documentation: https://dlang.org/phobos/core_internal_dassert.html
+ */
 module core.internal.dassert;
 
-/// Allows customized assert error messages for unary expressions
+/**
+ * Generates rich assert error messages for unary expressions
+ *
+ * The unary expression `assert(!una)` will be turned into
+ * `assert(!una, _d_assert_fail!"!"(una))`.
+ * This routine simply forward to the binary version as if the
+ * user wrote `assert(una == false)`.
```

Comment:
It doesn't call `_d_assert_fail` but rather produces the same string as `_d_assert_fail!"=="(una, false)`.